### DISCUSSION TITLE
Kraken ledger CSV data import

### DISCUSSION THREAD

**Discussion by fourofour404** [eprbell/dali-rp2#122] [Q&A] [answered]
Firstly, thank you for writing this software. 

I'd like to use dali-rp2 to import my ledger files from Kraken. Hence, I was looking into the data plugin files but could not find a plugin for this. As far as I understand the docs, the Kraken Locked CCXT pair converter plugin does not include data loading, but conversion only, right? 
After checking the plugins, the structure of ledger file in the trezor data loader seems quite similar to my downloaded Kraken ledger CSV. Would you suggest to modify this trezor data loader and adjust it to the Kraken structure in order to prepare a proper output file? 
Or is it better to adjust the manual CSV loader to that structure? Maybe such configuration is already available to you, then I'd be really glad if you could share it. 

The header of my Kraken CSV file has the following entries
`txid | refid | time | type | subtype | aclass | asset | amount | fee | balance`

Thank you for the support.

**Comment by eprbell** [comment, marked as answer]
Thanks! Correct: pair converter plugins only focus on price conversion. However take a look at #114: it's a Kraken REST plugin that is being worked on by @ndopencode. If you want you can also write a Kraken CSV plugin: the best way to do that is to use the Trezor plugin as boilerplate and adjust the format. Hope this helps.
